scanpy/external/tl/_annotator.py

```diff
@@ -0,0 +1,146 @@
+    genes from gene expression data with the Mann-Whitney U test, then annotate
+    them with the hypergeometric test, and finally filter out cell types that
+    have zero scores for all cells. The results are scores that tell how
+    probable is each cell type for each cell.
+
+    Parameters
+    ----------
+    adata
+        Tabular data with gene expressions.
+    markers
+        The data-frame with marker genes and cell types. Data-frame has two
+        columns **Gene** and **Cell Type** first holds gene names or ID and
+        second cell type for this gene. Gene names must be written in the same
+        format than genes in `adata`.
+    num_genes
+        The number of genes that the organism has.
+    return_nonzero_annotations
+        If true return scores only for cell types that have no zero scores.
+    p_threshold
+        A threshold for accepting the annotations. Annotations that have FDR
+        value bellow this threshold are used.
+    p_value_fun
+        A function that calculates a p-value. It can be either
+        `binom` that uses binom.sf or
+        `hypergeom` that uses hypergeom.sf.
+    z_threshold
+        The threshold for selecting the gene from gene expression data.
+        For each cell the attributes with z-value above this value are selected.
+    scoring
+        Scoring method for cell type scores. Available scores are:
+
+        exp_ratio
+            Proportion of genes typical for a cell type expressed in the cell
+        sum_of_expressed_markers
+            Sum of expressions of genes typical for a cell type
+        log_fdr
+            Negative of the logarithm of an false discovery rate (FDR) value
+        log_p_value
+            Negative of the logarithm of a p-value
+    normalize : bool, optional (default = False)
+        If this parameter is True data will be normalized during the
+        a process with a log CPM normalization.
+        That method works correctly data needs to be normalized.
+        Set this `normalize` on True if your data are not normalized already.
+
+    Returns
+    -------
+    pd.DataFrame
+        Cell type for each cell for each cell. The result is a sore matrix that
+        tells how probable is each cell type for each cell. Columns are cell
+        types and rows are cells.
+
+    Example
+    -------
+
+    Here is the example of annotation of dendritic cells based on their gene
+    expressions. For annotation, we use data by Villani et al. (2017)[1] and
+    marker genes by Franzén et al. (2019)[2].
+
+    [1]  Villani, A. C., Satija, ... Jardine, L. (2017). Single-cell
+         RNA-seq reveals new types of human blood dendritic cells, monocytes,
+         and progenitors. Science, 356(6335).
+
+    [2]  Oscar Franzén, Li-Ming Gan, Johan L M Björkegren, PanglaoDB:
+         a web server for exploration of mouse and human single-cell RNA
+         sequencing data, Database, Volume 2019, 2019.
```

Comment:
Please put the references into `references.rst` and reference them correctly, e.g. `[Villani17]_`

Reply:
Done